pringles/simulator/simulator.py

```diff
@@ -127,49 +128,52 @@ def get_process_output(self) -> str:
-
-    def discover_atomics(self) -> None:
-        assert self.user_models_dir is not None
-
-        files_to_extract_from = []
-        for filename in os.listdir(self.user_models_dir):
-            filename_with_path = os.path.join(self.user_models_dir, filename)
-            if os.path.isfile(filename_with_path):
-                _, file_extension = os.path.splitext(filename)
-                if file_extension in AtomicRegistry.SUPPORTED_FILE_EXTENSIONS:
-                    files_to_extract_from.append(filename_with_path)
-
-        # extract metadata from discovered source files
-        for discovered_path in files_to_extract_from:
-            with open(discovered_path, "r") as discovered_file:
-                try:
-                    discovered_metadata = AtomicMetadataExtractor(discovered_file).extract()
-                except MetadataParsingException:
-                    pass
-                else:
-                    atomic_class_builder = AtomicModelBuilder().with_name(discovered_metadata.name)
-                    for name in discovered_metadata.input_ports:
-                        atomic_class_builder.with_input_port(name)
-                    for name in discovered_metadata.output_ports:
-                        atomic_class_builder.with_output_port(name)
-                    built_class = atomic_class_builder.build()
-                    self._add_atomic_class_as_attribute(discovered_metadata.name,
-                                                        built_class)
-                    self.discovered_atomics.append(built_class)
+class Simulation:
+    DEFAULT_PICKLEFILE_NAME = 'simulation.pkl'
+
+    def __init__(self,
+                 top_model: Model,
+                 duration: Optional[VirtualTime] = None,
+                 events: Optional[List[Event]] = None,
+                 use_simulator_logs: bool = True,
+                 use_simulator_out: bool = True,
+                 working_dir: Optional[str] = None,
+                 override_logged_messages: Optional[str] = None):
+        self.result: Optional[SimulationResult] = None
+
+        self.top_model = top_model
+        self.duration = duration
+        self.events = events
+        self.use_simulator_logs = use_simulator_logs
+        self.use_simulator_out = use_simulator_out
+        self.override_logged_messages = override_logged_messages
+
+        self.working_dir = working_dir if working_dir else tempfile.mkdtemp()
+        self.output_dir = self.make_output_dir(self.working_dir)
+
+    @staticmethod
+    def make_output_dir(working_dir: str) -> str:
+        output_dir_name = datetime.now().strftime("%Y-%m-%d-%H%M%S") + "-" + str(uuid.uuid4().hex)
+        absolute_output_dir = os.path.join(working_dir, output_dir_name)
+        os.mkdir(absolute_output_dir)
+        return absolute_output_dir
+
+    @property
+    def was_executed(self) -> bool:
+        return self.result is not None
+
+    def to_pickle(self, path=None) -> None:
```

Comment:
`to_pickle` is some standard method? If not, we could use something more expressive, considering users that do not know what pickle is.

Reply:
That's the name of the methods in pandas.  It's `a_dataframe.to_pickle(filename)` and `pd.read_pickle(filename)` so it made sense to use the same ones

Reply:
Fair enough.